src/frontend/src/handler/create_table.rs

```diff
@@ -1032,7 +1032,34 @@ pub(super) async fn handle_create_table_plan(
 
                 let (columns, pk_names) = match wildcard_idx {
                     Some(_) => bind_cdc_table_schema_externally(cdc_with_options.clone()).await?,
-                    None => bind_cdc_table_schema(&column_defs, &constraints, None)?,
+                    None => {
+                        let (mut columns, pk_names) =
+                            bind_cdc_table_schema(&column_defs, &constraints, None)?;
+                        // read default value definition from external db
+                        let (options, secret_refs) = cdc_with_options.clone().into_parts();
+                        let config = ExternalTableConfig::try_from_btreemap(options, secret_refs)
+                            .context("failed to extract external table config")?;
+
+                        let table = ExternalTableImpl::connect(config)
```

Comment:
Maybe you can add a new type of `ExternalTableImpl` (`MockCDC`) and `CdcSourceType` (`mock-cdc`) only for testing purpose. This may help for other tests as well.

Reply:
Can you take a try to test it in the e2e test with the explain statment? With that we can also test this PR's logic (If optimizer read the right default column from PG)